### ISSUE TITLE
macOS build failure for 2018.01.01.00 and 2018.01.08.00

### ISSUE THREAD

**Issue by ilovezfs** [facebook/folly#729]
```
/bin/sh ./libtool  --tag=CXX   --mode=compile clang++ -DHAVE_CONFIG_H   -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include   -std=gnu++1y -g -O2 -c -o synchronization/LifoSem.lo synchronization/LifoSem.cpp
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c synchronization/LifoSem.cpp  -fno-common -DPIC -o synchronization/.libs/LifoSem.o
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c synchronization/AsymmetricMemoryBarrier.cpp -o synchronization/AsymmetricMemoryBarrier.o >/dev/null 2>&1
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c synchronization/LifoSem.cpp -o synchronization/LifoSem.o >/dev/null 2>&1
/bin/sh ./libtool  --tag=CXX   --mode=compile clang++ -DHAVE_CONFIG_H   -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include   -std=gnu++1y -g -O2 -c -o synchronization/ParkingLot.lo synchronization/ParkingLot.cpp
/bin/sh ./libtool  --tag=CXX   --mode=compile clang++ -DHAVE_CONFIG_H   -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include   -std=gnu++1y -g -O2 -c -o system/MemoryMapping.lo system/MemoryMapping.cpp
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c synchronization/ParkingLot.cpp  -fno-common -DPIC -o synchronization/.libs/ParkingLot.o
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c system/MemoryMapping.cpp  -fno-common -DPIC -o system/.libs/MemoryMapping.o
/bin/sh ./libtool  --tag=CXX   --mode=compile clang++ -DHAVE_CONFIG_H   -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include   -std=gnu++1y -g -O2 -c -o system/Shell.lo system/Shell.cpp
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c system/Shell.cpp  -fno-common -DPIC -o system/.libs/Shell.o
In file included from synchronization/ParkingLot.cpp:16:
In file included from ./../folly/synchronization/ParkingLot.h:25:
./../folly/Indestructible.h:154:7: error: implicit instantiation of undefined template 'std::__1::array<folly::parking_lot_detail::Bucket, 4096>'
    T value;
      ^
./../folly/Indestructible.h:167:11: note: in instantiation of member class 'folly::Indestructible<std::__1::array<folly::parking_lot_detail::Bucket, 4096> >::Storage' requested here
  Storage storage_{};
          ^
synchronization/ParkingLot.cpp:27:58: note: in instantiation of template class 'folly::Indestructible<std::__1::array<folly::parking_lot_detail::Bucket, 4096> >' requested here
  static Indestructible<std::array<Bucket, kNumBuckets>> gBuckets;
                                                         ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tuple:222:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
1 error generated.
make[2]: *** [synchronization/ParkingLot.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c system/Shell.cpp -o system/Shell.o >/dev/null 2>&1
libtool: compile:  clang++ -DHAVE_CONFIG_H -I./.. -pthread -I/usr/local/include -I/usr/local/Cellar/openssl/1.0.2n/include -std=gnu++1y -g -O2 -c system/MemoryMapping.cpp -o system/MemoryMapping.o >/dev/null 2>&1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

**Comment by ilovezfs** [comment]
CC @Orvid @simpkins @wez this is still a problem in folly 2018.01.08.00

Ref https://github.com/Homebrew/homebrew-core/pull/22553

**Comment by ilovezfs** [comment]
This looks like it's fixed as of 2018.01.15.00
